video/out/context_wayland.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of mpv video player.
+ * Copyright © 2013 Alexander Preisinger <alexander.preisinger@gmail.com>
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "video/out/wayland_common.h"
+#include "opengl/context.h"
+
+
+struct priv {
+
+};
+
+static void wayland_uninit(struct ra_ctx *ctx)
+{
+    ra_free(&ctx->ra);
+    vo_wayland_uninit(ctx->vo);
+}
+
+static bool wayland_init(struct ra_ctx *ctx)
+{
+	if (!vo_wayland_init(ctx->vo)) {
+        vo_wayland_uninit(ctx->vo);
+        return false;
+    }
+
+    ctx->ra =  talloc_zero(NULL, struct ra);
```

Comment:
I'd say it's worth filling in the ra fns with trivial functions to avoid adding the conditionals in the hwdecs.